libtiff/libtiff_ctypes.py

```diff
@@ -524,24 +524,31 @@ def get_tag_define(tagname):
     @classmethod
     def open(cls, filename, mode='r'):
         """ Open tiff file as TIFF.
+
+        Parameters
+        ----------
+        filename: str or bytes.
+          The path to the file.
+        mode: str
+          Specifies if the file is to be opened for reading ('r'), writing ('w'),
+          or appending ('a'). Optional flags can be passed. See the documentation
+          of TIFFOpen() for the complete list.
         """
-        try:
+        if isinstance(filename, str) and hasattr(libtiff, "TIFFOpenW"):
+            # On Windows, the only reliable way to open a file with unicode characters
+            # is to use the *W function.
+            tiff = libtiff.TIFFOpenW(filename, mode.encode('ascii'))
+        else:
+            # Python3: it needs bytes for the arguments of type "c_char_p"
             try:
-                # Python3: it needs bytes for the arguments of type "c_char_p"
-                if isinstance(filename, str) and 'win' in sys.platform:
-                    filename = filename.encode(locale.getpreferredencoding(False))
-                else:
-                    filename = os.fsencode(filename)  # no-op if already bytes
-            except AttributeError:
-                # Python2: it needs str for the arguments of type "c_char_p"
-                if isinstance(filename, unicode):  # noqa: F821
-                    filename = filename.encode(sys.getfilesystemencoding())
-        except Exception as ex:
-            # It's probably going to not work, but let it try
-            print('Warning: filename argument is of wrong type or encoding: %s'
-                  % ex)
-
-        tiff = libtiff.TIFFOpen(filename, mode.encode('ascii'))
+                filename = os.fsencode(filename)  # no-op if already bytes
+            except Exception as ex:
+                # It's probably going to not work, but let it try
+                print('Warning: filename argument is of wrong type or encoding: %s'
+                      % ex)
```

Comment:
Personal preference:

```suggestion
                print(f"Warning: filename argument is of wrong type or encoding: {ex}")
```

Reply:
Or... @pearu thoughts on `warnings.warn`?

Reply:
Yes, using `warnings.warn` is preferable.

Reply:
I'll happily execute... Honestly, I'd love removing all the `print()`. They should either to logging or to warnings.warn (or even to an Exception).
Our software is a GUI, and when using this library, all the print's are essentially sent to nowhere. Using logging is much easier to redirect them to a log file.